### ISSUE TITLE
data preprocessing

### ISSUE THREAD

**Issue by Lj4040** [grammarly/gector#177]
What do SOURCE and TARGER stand for in data preprocessing? Could you explain them? Thank you for your reply

**Comment by skurzhanskyi** [comment]
As [it was mentioned ](https://github.com/grammarly/gector/issues/136), source is original text, target is corrected text

**Comment by Lj4040** [comment]
For example, what I downloaded is the FCE data set, which contains M2 file and json file. In this file, there is no distinction between correct and incorrect sentences. How should I pass the data processing file.I would like to ask for your guidance, for which I greatly appreciate it

**Comment by Lj4040** [comment]
Only the downloaded synthetic data set has correct and incorrect sentences, do we have to use the synthetic data to pass in?

**Comment by skurzhanskyi** [comment]
You can take a look at the [M2scorer](https://github.com/nusnlp/m2scorer) repository and specifically the [edit_creator.py](https://github.com/nusnlp/m2scorer/blob/master/scripts/edit_creator.py) script. 
To get original/source sentences, you can simply run `cat myfile.m2 | grep "^S " | cut -c3-  > myfile.src`

**Comment by Lj4040** [comment]
Sincerely thank you for your answer, I will try

**Comment by Lj4040** [comment]
Dear author, after data processing, the data set has become the following picture. This format file is quite different from the M2 file, so I'm not sure if it's correct.May I ask if the data set in this format is correct?
![图片1](https://user-images.githubusercontent.com/117823125/209660353-3d3665b5-b742-44a6-ac4e-125318baa6c6.png)
Because I am a beginner of GEC, some questions may be a little naive. I hope you can understand. Thank you for your reply

**Comment by skurzhanskyi** [comment]
Yes, this is a specific format for training to save only input tokens and corresponding tags
